CONTRIBUTING.md

```diff
@@ -1,221 +1,295 @@
 
-A `Graphic` is something that can be added to a `PlotArea` (described in detail in a later section). All the various 
-fastplotlib graphics, such as `ImageGraphic`, `ScatterGraphic`, etc. inherit from the `Graphic` base class in 
-`fastplotlib/graphics/_base.py`. It has a few properties that mostly wrap `pygfx` `WorldObject` properties and transforms. 
-These might change in the future (ex. `Graphic.position_x` etc.).
+Next, we will be notified of the pull request and will read it over. We will try to give and initial response quickly, and then do a longer in-depth
+review, at which point you will probably need to respond to our comments, making changes as appropriate. We will then respond again, and proceed
+in an iterative fashion until everyone is happy with the proposed changes.
 
-All graphics can be given a string name for the user's convenience. This allows graphics to be easily accessed from 
-plots, ex: `subplot["some_image"]`.
+Once your changes are integrated, you will be added as a GitHub contributor. Thank you for being 
+apart of `fastplotlib`!
 
-All graphics contain a `world_object` property which is just the `pygfx.WorldObject` that this graphic uses. Fastplotlib 
-keeps a *private* global dictionary of all `WorldObject` instances and users are only given a weakref proxy to this world object. 
-This is due to garbage collection. This may be quite complicated for beginners, for more details see this PR: https://github.com/fastplotlib/fastplotlib/pull/160 . 
-If you are curious or have more questions on garbage collection in fastplotlib you're welcome to post an issue :D. 
+### Style Guide
 
-#### Graphic properties
+As far as code style, please adhere to the following guidelines:
 
-Graphic properties are all evented, and internally we called these "graphic features". They are the various 
-aspects of a graphic that the user can change.
-The "graphic features" subpackage can be found at `fastplotlib/graphics/_features`. As we can see this 
-is a private subpackage and never meant to be accessible to users..
+- Longer, descriptive names are preferred (e.g., `x` is not an appropriate name for a variable), especially for anything user-facing,
+such as methods, attributes, or arguments
+- Any public method or function must have complete type-annotated docstrings (see below for details). Hidden ones do not need to have
+complete docstring, but they probably should.
 
-##### LineGraphic
+### Releases
 
-For example let's look at `LineGraphic` in `fastplotlib/graphics/line.py`. Every graphic has a class variable called 
-`_features` which is a set of all graphic properties that are evented. It has the following evented properties: 
-`"data", "colors", "cmap", "thickness"` in addition to properties common to all graphics, such as `"name", "offset", "rotation", and "visible"` 
+We create releases on GitHub, deploy on / distribute via [pypi](https://pypi.org/), and try to follow [semantic versioning](https://semver.org/):
```

Comment:
```suggestion
We create releases on GitHub and distribute via [pypi](https://pypi.org/), and try to follow [semantic versioning](https://semver.org/):
```

Reply:
We're currently in alpha so we're not exactly following this, we're still at zero version